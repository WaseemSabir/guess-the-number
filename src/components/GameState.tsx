import React from 'react';

interface GameStateProps {
    lower: string,
    upper: string,
    lastGuess: string,
    status: string | null
}

function GameState(props: GameStateProps) {
    const {lower, upper, lastGuess, status} = props

    return (
        <>
            <h3>Play!</h3>
            <div>Guess the number between {lower} and {upper}</div>
            <div>Last guess: {lastGuess}</div>
            {
                status && <div>{status}</div>
            }
        </>
    );
}

export default GameState;
