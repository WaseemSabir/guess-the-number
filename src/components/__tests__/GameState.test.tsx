import {cleanup, render} from "@testing-library/react";
import React from "react";
import renderer from "react-test-renderer";
import GameState from "../GameState";


afterEach(cleanup)

it('should render without crashes', () => {
    const {unmount} = render(getGameState());
    unmount();
});

it('should match the last snapshot', () => {
    const component = renderer.create(
        getGameState()
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

function getGameState() {
    return (
        < GameState lower={"10"} upper={"100"} lastGuess={"9"} status={"You got it!"}/>
    );
}
