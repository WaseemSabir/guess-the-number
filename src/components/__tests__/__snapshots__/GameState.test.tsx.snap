// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the last snapshot 1`] = `
Array [
  <h3>
    Play!
  </h3>,
  <div>
    Guess the number between 
    10
     and 
    100
  </div>,
  <div>
    Last guess: 
    9
  </div>,
  <div>
    You got it!
  </div>,
]
`;
